what4/src/What4/Expr/Builder.hs

```diff
@@ -4804,6 +4823,62 @@ instance IsExprBuilder (ExprBuilder t st fs) where
    | otherwise = do
      sbMakeExpr sym $ BVAshr (bvWidth x) x y
 
+  bvRol sym x y
+   | Just i <- asUnsignedBV x, Just n <- asUnsignedBV y
+   = bvLit sym (bvWidth x) $ rotateLeft (bvWidth x) i n
+
+   | Just n <- asUnsignedBV y
+   , n `rem` intValue (bvWidth x) == 0
+   = return x
+
+   | Just (BVRol w x' n) <- asApp x
+     -- NB 0 <= n < w
+   = do wbv <- bvLit sym w (intValue w)
+        y' <- bvUrem sym y wbv
```

Comment:
I'd like to avoid doing this `bvUrem` in the common case when `w` is a power of 2, so that when you have more than two successive rotates all the additions can combine together. (Unless there are some other rewrite rules that can simplify away the redundant `bvUrem` calls in this situation, but I don't expect there are.)